fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexScrubbing.java

```diff
@@ -228,14 +233,50 @@ private void reportIssues(List<IndexScrubbingTools.Issue> issueList, Throwable e
     IndexingRangeSet getRangeset(FDBRecordStore store, Index index) {
         switch (scrubbingType) {
             case MISSING:
-                return IndexingRangeSet.forScrubbingRecords(store, index);
+                return IndexingRangeSet.forScrubbingRecords(store, index, scrubbingPolicy.getRangeId());
             case DANGLING:
-                return IndexingRangeSet.forScrubbingIndex(store, index);
+                return IndexingRangeSet.forScrubbingIndex(store, index, scrubbingPolicy.getRangeId());
```

Comment:
Would it make sense to decouple the scrubbing type from the range id?
i.e. to have this method be something like:
```
if (scrubbingPolicy.getRangeId() == 0) {
    switch (scrubbingType) {
    case MISSING:
        return IndexingRangeSet.forScrubbingRecords(store, index);
        return IndexingRangeSet.forScrubbingRecords(store, index, scrubbingPolicy.getRangeId());
    case DANGLING:
        return IndexingRangeSet.forScrubbingIndex(store, index);
        return IndexingRangeSet.forScrubbingIndex(store, index, scrubbingPolicy.getRangeId());
    default:
        throw new RecordCoreArgumentException("Unpredicted scrubbing type ");
    }
} else {
    return IndexingRangeSet.forScrubbingId(store, scrubbingPolicy.getRange());
}
```